src/test/java/com/puppycrawl/tools/checkstyle/internal/utils/XdocGenerator.java

```diff
@@ -53,21 +61,30 @@ public static void generateXdocContent() throws Exception {
         for (Path path : templatesFilePaths) {
             final String pathToFile = path.toString();
             final File inputFile = new File(pathToFile);
-            final File outputFile = new File(pathToFile.replace(".template", ""));
-
+            final File tempFile = File.createTempFile(pathToFile.replace(".template", ""), "");
             final XdocsTemplateSinkFactory sinkFactory = (XdocsTemplateSinkFactory)
                     plexus.lookup(SinkFactory.ROLE, XDOCS_TEMPLATE_HINT);
-            final Sink sink = sinkFactory.createSink(outputFile.getParentFile(),
-                    outputFile.getName(), String.valueOf(StandardCharsets.UTF_8));
+            final Sink sink = sinkFactory.createSink(tempFile.getParentFile(),
+                    tempFile.getName(), String.valueOf(StandardCharsets.UTF_8));
             final XdocsTemplateParser parser = (XdocsTemplateParser)
                     plexus.lookup(Parser.ROLE, XDOCS_TEMPLATE_HINT);
             try (Reader reader = ReaderFactory.newReader(inputFile,
                     String.valueOf(StandardCharsets.UTF_8))) {
                 parser.parse(reader, sink);
             }
             finally {
+
                 sink.close();
+
+                final File outputFile = new File(pathToFile.replace(".template", ""));
+
```

Comment:
Please move code to copy file to destination out of finally

Reply:
Done @romani